### PR TITLE
feat(scheduling): [MC-523] Curators must provide a reason for manuall…  …y scheduling items

### PR DESCRIPTION
**Pull request by nina-py**
## Goal

- Added reasons for manually scheduling items + comment field for "Other" reasons to scheduling form. These only appear if requested, and are required if they appear on the page.

- Reorganised the elements on the Scheduling modal, so that surface and date are now on a single line, and topic & publisher distribution is rolled up by default in an accordion element.

- Made sure manually added items from the Prospecting page also have a list of reasons to choose from, identical to the Scheduling page.

- Added tests.

## Screenshots

The reworked schedule modal:

reworked-schedule-modal

For other surfaces, it looks more compact without the reasons:

schedule-modal-non-us

Expanded topic & publisher distribution:

schedule-modal-non-us-expanded

## Video walkthrough

https://github.com/Pocket/curation-admin-tools/assets/22447785/26300366-db36-403a-a1cb-23bec4fe49a7

## Reference

https://mozilla-hub.atlassian.net/browse/MC-523

### CODE REVIEW

File: src/curated-corpus/components/RemoveItemForm/RemoveItemForm.validation.tsx [hunk 1]
```diff
@@ -22,7 +22,7 @@ export const validationSchema = yup
     ['OTHER']: yup.boolean(),
     reasonComment: yup
       .string()
-      .max(100, 'Reason is too long, cannot exceed 100 characters.'), // max 50 chars for now
+      .max(100, 'Reason is too long, cannot exceed 100 characters.'), // max 100 chars for now
```
Comment (nina-py) [inline src/curated-corpus/components/RemoveItemForm/RemoveItemForm.validation.tsx:25]: Typo fix, unrelated to PR

File: src/curated-corpus/components/ScheduleItemForm/ScheduleItemForm.test.tsx [hunk 2]
```diff
@@ -156,4 +160,111 @@ describe('The ScheduleItemForm component', () => {
...
+    });
+
+    const errorMessage = screen.getByText(
+      /Please choose at least one reason to schedule this item manually./i
+    );
+    expect(errorMessage).toBeInTheDocument();
+    expect(handleSubmit).not.toHaveBeenCalled();
+  });
+
+  it.todo(
```
Comment (nina-py) [inline src/curated-corpus/components/ScheduleItemForm/ScheduleItemForm.test.tsx:237]: 😭

File: src/curated-corpus/components/ScheduleItemForm/ScheduleItemForm.tsx [hunk 3]
```diff
@@ -92,20 +119,53 @@ export const ScheduleItemForm: React.FC<
...
+      [ManualScheduleReason.PublisherDiversity]: false,
+      [ManualScheduleReason.TimeSensitiveExplainer]: false,
+      [ManualScheduleReason.TimeSensitiveNews]: false,
+      [ManualScheduleReason.TopicDiversity]: false,
+      [ManualScheduleReason.Trending]: false,
+      [ManualScheduleReason.UnderTheRadar]: false,
+      OTHER: false,
+      manualScheduleReason: '',
+      reasonComment: '',
     },
```
Comment (nina-py) [inline src/curated-corpus/components/ScheduleItemForm/ScheduleItemForm.tsx:147]: Credit to @katerinachinnappan for a nice pattern I was able to reuse here 🚀

**Review by jpetto — DISMISSED** [review]
only the flaky time-based test is failing and this lgtm, so :shipit:

**Review by jpetto — APPROVED** [review]

**Pull request merged**